app/compare/diff_chunk.tsx

```diff
@@ -1,61 +1,113 @@
+import DiffMatchPatch from "diff-match-patch";
 import React from "react";
-import JsDiff from "diff";
 import { OutlinedButton } from "../components/button/button";
 
 export type DiffChunkProps = {
-  change: JsDiff.Change;
+  change: DiffMatchPatch.Diff;
   defaultExpanded: boolean;
 };
 
-type DiffChunkState = {
+type State = {
   expanded: boolean;
 };
 
-export default class DiffChunk extends React.Component<DiffChunkProps, DiffChunkState> {
-  state = { expanded: this.props.defaultExpanded };
+const MIN_COLLAPSED_UNCHANGED_REGION_SIZE = 16;
+const MIN_COLLAPSED_CHANGED_REGION_SIZE = 128;
```

Comment:
what does "size" mean here? is it number of lines?

size in my mind is typically {Width, Height}

Reply:
Makes sense; replaced "SIZE" with "NUM_LINES"